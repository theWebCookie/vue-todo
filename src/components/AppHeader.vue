<template>
  <header :class="$style.header"></header>
</template>

<script>
export default {
  name: 'AppHeader',
  components: {},
};
</script>

<style module>
.header {
  position: relative;
  background-image: url('../assets//header.jpg');
  background-repeat: no-repeat;
  background-position: -250px -370px;
  width: 100%;
  height: 450px;
}

.header::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  background: linear-gradient(to right, rgba(168, 0, 168, 0.7), rgba(0, 51, 161, 0.7));
  width: 100%;
  height: 450px;
}
</style>
