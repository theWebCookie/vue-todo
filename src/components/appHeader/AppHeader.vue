<template>
  <header :class="$style.header"></header>
</template>

<script>
export default {
  name: 'AppHeader',
  components: {},
};
</script>

<style module lang="scss">
.header {
  position: relative;
  background-image: url('../../assets/header.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  width: 100%;
  height: 450px;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    background: linear-gradient(to right, rgba(168, 0, 168, 0.7), rgba(0, 51, 161, 0.7));
    width: 100%;
    height: 450px;
  }
}
</style>
