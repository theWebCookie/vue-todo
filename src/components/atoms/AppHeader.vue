<template>
  <header :class="$style.header"></header>
</template>

<script>
export default {
  name: 'AppHeader',
};
</script>

<style module lang="scss">
.header {
  position: relative;
  background-image: url('../../assets/header.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 80%;
  width: 100%;
  height: 450px;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    background: linear-gradient($header-gradient-angle, $header-gradient-start, $header-gradient-middle, $header-gradient-end);
    width: 100%;
    height: 450px;
  }
}
</style>
