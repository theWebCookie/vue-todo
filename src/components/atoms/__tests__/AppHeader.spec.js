import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from '@jest/globals';
import AppHeader from '../AppHeader.vue';

describe('AppHeader.vue', () => {
  it('renders correctly and matches the snapshot', () => {
    const $style = {
      header: 'header',
    };

    const wrapper = shallowMount(AppHeader, {
      mocks: {
        $style,
      },
    });

    const headerElement = wrapper.find('header');
    expect(headerElement.exists()).toBe(true);
    expect(headerElement.classes()).toContain($style.header);
    expect(wrapper.element).toMatchSnapshot();
  });
});
