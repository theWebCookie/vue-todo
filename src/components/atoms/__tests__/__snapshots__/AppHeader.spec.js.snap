// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.vue renders correctly and matches the snapshot 1`] = `
<header
  class="header"
/>
`;
